### DISCUSSION TITLE
Checkpoint mod

### DISCUSSION THREAD

**Discussion by lilyremigia** [ppy/osu#13676] [Ideas]
As an alternative to no-fail mod:
When the user fails they get reverted back to the previous break (or start of the map). The end result would probably be the passes sliced together. Probably should have a huge negative modifier.
The mod could also be combined with sudden death or perfect to achieve a sliced perfect play.

**Comment by LittleEndu** [comment]
Breaks aren't distributed evenly across maps. Some maps have many breaks, some have none. Such a mod should probably use some other metric, like note density, to evenly distribute X number of checkpoints.

**Reply by lilyremigia** [reply]
The reason why I would put the checkpointss at breaks, since at breaks it doesn't matter where your cursor is.

**Reply by LittleEndu** [reply]
Sure, but you can make use of what pause does, the "place your cursor here to continue" function
